### ISSUE TITLE
Dropdown navigation with arrow keys not working

### ISSUE THREAD

**Issue by bbinoy** [reactstrap/reactstrap#580]
### Issue description

- version 4.8.0
- components: Dropdown

Dropdown items are not selectable by using arrow keys, which is [supported by default by bootstrap v4 beta](https://getbootstrap.com/docs/4.0/components/dropdowns/).

Also closing the dropdown with the ESC key is not working, contrary to bootstrap default.

**Comment by toodlooo** [comment]
Is this feature supposed to work for Custom Dropdowns? It looks like it works for all dropdown types except Custom: https://reactstrap.github.io/components/dropdowns/

**Comment by TheSharpieOne** [comment]
@toodlooo You are sorta correct, it will not work for dropdowns which do not have any dropdown items (defined as `.dropdown-item`) which `DropdownItem` adds for you.

This is because we select the items via https://github.com/reactstrap/reactstrap/blob/c89bdea5ca7b90641708438c2415463996f2ab76/src/Dropdown.js#L132
Since not all of the elements within a dropdown menu are "items" (some can be dividers or grouping labels or something) we cannot just use all of the child elements.

Maybe we need a better convention so that the selector is not the same thing/class that applies styles so that the dropdown items can look different and also be selected.

It's also worth noting that in the example it manually toggles the menu when an item "selected" as `DropdownItem` would do that for you and not using `DropdownItem` means you have to do that yourself

**Comment by alexw23** [comment]
Looks like this is still happening https://reactstrap.github.io/components/dropdowns/#app
